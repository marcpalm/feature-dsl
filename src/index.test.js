const path = require('path')
const fs = require('fs')

const {
    printFeature,
    parse,
 } = require('./index.js')

describe('printFeature', () => {
    it('should write file as parsed', () => {
        var file = fs.readFileSync(path.join( __dirname, '..', './samples/sample.feature'))
        const feature = parse(file)
        expect(feature).toMatchSnapshot()

        expect(printFeature(feature[0])).toMatch(file.toString())
    })

})