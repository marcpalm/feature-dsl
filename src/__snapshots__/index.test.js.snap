// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printFeature should write file as parsed 1`] = `
Array [
  Object {
    "desire": "to take beer off the wall",
    "feature": "Can drink beer when thirsty",
    "perspective": "a drinker",
    "reason": "to satisfy my thirst",
    "scenarios": Array [
      Object {
        "given": Array [
          "100 bottles of beer on the wall",
        ],
        "scenario": "Can take a single beer",
        "then": Array [
          "there are 98 bottles of beer on the wall",
        ],
        "when": Array [
          "a bottle is taken down",
          "a bottle is taken down",
        ],
      },
      Object {
        "given": Array [
          "100 bottles of beer on the wall",
        ],
        "scenario": "Can take multiple beers",
        "then": Array [
          "there are 95 bottles of beer on the wall",
        ],
        "when": Array [
          "5 bottles are taken down",
        ],
      },
      Object {
        "given": Array [
          "100 bottles of beer on the wall",
          "there is nobody in the room",
        ],
        "scenario": "Ghosts can drink",
        "then": Array [
          "there are 95 bottles of beer on the wall",
          "there are ghosts in the room",
        ],
        "when": Array [
          "5 bottles are taken down",
          "they are floating in the air",
        ],
      },
    ],
  },
]
`;
